#8 Swagger setup

Skybby15 opened the pull request:
Swagger runs on http://localhost:8080/swagger-ui.html , (the ".html" was missing and i could not open it otherwise) 
Added openApi to dependencies

Skybby15 committed:
Swagger setup on Backend + Frontend
Dependency added on Backend
DTO and Service generated and Script added on Frontend

mmihailescu98 commented on frontend/src/app/api/api.module.ts:
Fisierele de genul asta, care sunt generate automat nu trebuie adaugate. Ma gandesc ca erau sub "unversioned files". Pe partea de FE ar trebui dat push doar cu ce e in package.json 

Overall you should just push 4 files I think -> for BE the controller with the new endpoint for testing that, the PetDTO, and for FE the package.json

Skybby15 replied:
Am adaugat in .gitignore din frontend intreg folderul "api" (/src/app/api/) , si fisierul openapitools.json , ambele create de comanda de generare

mmihailescu98 commented on backend/src/main/java/cloudflight/integra/backend/SwaggerTestModel/SwagPetDTO.java:
Package name-ul ar trebui sa fie dto, dar in rest este bine facut recordul

Skybby15 replied:
L-am redenumit

mmihailescu98 commented on frontend/package.json:
```diff
@@ -28,6 +29,7 @@
     "@angular/forms": "^20.2.0",
     "@angular/platform-browser": "^20.2.0",
     "@angular/router": "^20.2.0",
+    "@openapitools/openapi-generator-cli": "^2.23.1",
```
Should be moved to devDependencies since we only need it for development

Skybby15 replied:
Am reinstalat dependency-ul sub devDependencies

mmihailescu98 commented on backend/build.gradle:
```diff
@@ -21,6 +21,9 @@ repositories {
 	implementation 'org.springframework.boot:spring-boot-starter'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
+
+    implementation 'org.thingsboard:springdoc-openapi-starter-webmvc-ui:2.8.8TB'
```
I would go with the "official" org.springdoc instead of trhingsboard. They should behave the same but I don't think we need the thingsboard one .

Skybby15 replied:
Am gasit si inlocuit cu cel oficial

Skybby15 committed:
Swagger setup requested changes
-Frontend generated api files untracked,
-Backend openApi dependency changed and renamed dto s package
-Frontend moved openApi to devDependencies

mmihailescu98 commented on backend/build.gradle:
```diff
@@ -21,6 +21,10 @@ repositories {
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
+
```
This comment is not needed and also the extra new lines, but I ll leave them for now.
